### PR TITLE
Add --help_markdown flag to generate documentation for the flags

### PR DESCRIPTION
**Pull request by ChadKillingsworth**
I wanted a page to document the flags. I added a `--help_markdown` flag to generate the flag usage in a format for the wiki. The output of that flag can be directly pasted into https://github.com/google/closure-compiler/wiki/Flags-and-Options

### CODE REVIEW

File: src/com/google/javascript/jscomp/CommandLineRunner.java [hunk 1]
```diff
@@ -934,12 +943,24 @@ private void printUsage(PrintStream ps) {
...
+            suffix =
+                "\n"
+                    + boldPrefix
+                    + "Available Error Groups: "
+                    + normalPrefix
+                    + DiagnosticGroups.DIAGNOSTIC_GROUP_NAMES;
+          }
+        }
+
+        if (this.helpMarkdown) {
```
[the remark this answers is not in the document]
Comment (ChadKillingsworth) [reply src/com/google/javascript/jscomp/CommandLineRunner.java]: Done

**Comment by brad4d** [comment]
imported and sent for internal testing and to @lauraharker for review

**Comment by brad4d** [comment]
Hit a snag.

This PR uses an args4j feature (`parser.getProperties()`) that doesn't exist in args4j 2.0.26,
which is the canonical version used internally at Google.
(Upgrading args4j would require someone to fix all projects that use args4j inside Google,
so not in scope for this PR.)

Our POM was incorrectly changed to use args4j version 2.33.
I'll need to undo that, and this PR needs to be changed to avoid any features not in
args4j 2.0.26.

Sorry.

**Comment by ChadKillingsworth** [comment]
That's an easy fix on my side when that's changed.

**Comment by brad4d** [comment]
args4j version change submitted internally
it'll be in tomorrow's push

**Comment by ChadKillingsworth** [comment]
Changes made @brad4d

**Comment by brad4d** [comment]
updates imported and sent for internal testing and review with @lauraharker

**Pull request closed without merging**